applications/luci-app-throughput-measurements/root/usr/libexec/rpcd/luci.tputmeas

```diff
@@ -0,0 +1,238 @@
+					end
+
+					iperf3_cmd = iperf3_cmd .. " " .. args.server_address
+				end
+
+				if  args.interface and args.interface ~= '' then
+					local ifa = nixio.getifaddrs()
+					for k, v in pairs(ifa) do
+						if ifa[k].name == args.interface and ifa[k].family == "inet" and ifa[k].addr then
+							iperf3_cmd = iperf3_cmd .. " -B " .. ifa[k].addr
+						end
+					end
+				elseif args.interface_address and args.interface_address ~= '' then
+					local ifa = nixio.getifaddrs()
+					local ok = false
+					for k, v in pairs(ifa) do
+						if ifa[k].addr == args.interface_address then
+							iperf3_cmd = iperf3_cmd .. " -B " .. ifa[k].addr
+							ok = true
+						end
+					end
+					if not ok then
+						return { result = "error: Incorrect format of an interface address" }
+					end
+				end
+
+				if args.port and args.port ~= '' then
+					if tonumber(args.port) then
+						iperf3_cmd = iperf3_cmd .. " -p " .. args.port
+					else
+						return { result = "error: Incorrect port argument" }
+					end
+				end
+
+				if args.duration and args.duration ~= '' then
+					if tonumber(args.duration) then
+						iperf3_cmd = iperf3_cmd .. " -t " .. args.duration
+					else
+						return { result = "error: Incorrect duration argument" }
+					end
+				end
+
+				if args.bytes and args.bytes ~= '' then
+					if args.bytes:match("^[0-9]*$") or args.bytes:match("^[0-9]*[kKmM]$") then
+						iperf3_cmd = iperf3_cmd .. " -n " .. args.bytes
+					else
+						return { result = "error: Incorrect bytes argument" }
+					end
+				end
+
+				if args.udp == "1" and args.udp ~= '' then
+					iperf3_cmd = iperf3_cmd .. " -u"
+				end
+
+				if args.bitrate and args.bitrate ~= '' then
+					if args.bitrate:match("^[0-9]*$") or args.bitrate:match("^[0-9]*[kKmM]$") then
+						iperf3_cmd = iperf3_cmd .. " -b " .. args.bitrate
+					else
+						return { result = "error: Incorrect bitrate argument" }
+					end
+				end
+
+				if args.reverse == "1" and args.reverse ~= nil then
+					iperf3_cmd = iperf3_cmd .. " -R"
+				end
+
+				if args.connections and args.connections ~= ''  then
+					if tonumber(args.connections) then
+						iperf3_cmd = iperf3_cmd .. " -P " .. args.connections
+					else
+						return { result = "error: Incorrect connections argument" }
+					end
+				end
+
+				if args.interval and args.interval ~= '' then
+					if tonumber(args.interval) then
+						iperf3_cmd = iperf3_cmd .. " -i " .. args.interval
+					else
+						return { result = "error: Incorrect interval argument" }
+					end
+				end
+			end
+
+			iperf3_cmd = iperf3_cmd .. " 2>&1"
+
+			local pid = posix.fork()
+			if pid == 0 then
+				local cpid = posix.getpid()
+				fs.writefile("/var/run/iperf3.pid",cpid)
+
+				local conn = ubus.connect()
+				if not conn then
+					print("Failed to connect to ubus")
+					os.exit(1)
+				end
+
+				local ubus_objects = {
+					["luci.tputmeas.notify"] = {
+					}
+				}
+				conn:add( ubus_objects )
+
+				sys.call("sleep 1")
```

Comment:
You can use `nixio.poll({}, 1000)` instead to avoid executing a shell / external sleep here.